sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java

```diff
@@ -146,6 +147,15 @@ public TableServiceAsyncClient buildAsyncClient() {
 
         AzureNamedKeyCredential namedKeyCredential = null;
 
+        // If 'endpoint' was provided, validate its format to end with the appropriate suffix
+        if (endpoint != null) {
+            String trimmedEndpoint = endpoint.endsWith("/") ? endpoint.substring(0,endpoint.length()-1) : endpoint;
+            if (!trimmedEndpoint.endsWith(suffix)) {
+                throw logger.logExceptionAsError(new IllegalArgumentException("The 'endpoint' provided is not valid." +
+                    " Please try re-creating the TableServiceClient with just the account Uri."));
```

Comment:
Sometimes people just focus on the error message and take a bit to see where it originated from, so I felt this was a bit more explicit on that front.
```suggestion
                throw logger.logExceptionAsError(
                    new IllegalArgumentException("Cannot build a TableServiceClient, the 'endpoint' provided is not " +
                        "valid. Please try again with using just the account URI."));
```

Reply:
We could also change the exception to be thrown in the `endpoint()` method or in the client constructor, although I'm not sure which of the 3 options would be preferred. Any suggestions @srnagar?